lib/_stream_readable.js

```diff
@@ -695,17 +707,27 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     debug('ondata');
     const ret = dest.write(chunk);
     debug('dest.write', ret);
-    if (ret === false) {
+    if (ret === false && !cleanedUp) {
       // If the user unpiped during `dest.write()`, it is possible
       // to get stuck in a permanently paused state if that write
       // also returned false.
       // => Check whether `dest` is still a piping destination.
-      if (((state.pipesCount === 1 && state.pipes === dest) ||
-           (state.pipesCount > 1 && state.pipes.includes(dest))) &&
-          !cleanedUp) {
-        debug('false write response, pause', state.awaitDrain);
-        state.awaitDrain++;
+
+      if ((state.pipesCount === 1 && state.pipes === dest)) {
+        debug('false write response, pause', 0);
+        state.awaitDrainWriters = dest;
+        state.multiAwaitDrain = false;
+      } else if ((state.pipesCount > 1 && state.pipes.includes(dest))) {
+        if (!state.multiAwaitDrain) {
+          state.multiAwaitDrain = true;
+          state.awaitDrainWriters = new Set(
+            state.awaitDrainWriters ? [state.awaitDrainWriters] : []
```

Comment:
Would it be possible to handle the else case here similar to the `pipesCount === 1` case?

Reply:
Do you mean like this: https://github.com/nodejs/node/pull/27572/commits/66d566fd5b65f9f4201dfe6d8184f276269ae30e#diff-ba6a0df0f5212f5cba5ca5179e209a17